src/main/java/pl/javastart/devicerent/app/ApplicationController.java

```diff
@@ -0,0 +1,80 @@
+package pl.javastart.devicerent.app;
+
+import org.springframework.stereotype.Controller;
+import pl.javastart.devicerent.components.category.CategoryController;
+import pl.javastart.devicerent.components.customer.CustomerController;
+import pl.javastart.devicerent.components.device.DeviceController;
+import pl.javastart.devicerent.components.rent.RentController;
+
+import java.util.Arrays;
+import java.util.Scanner;
+
+@Controller
+public class ApplicationController {
+    private Scanner scanner;
+    private ConsoleLogger logger;
+    private CategoryController categoryController;
+    private CustomerController customerController;
+    private DeviceController deviceController;
+    private RentController rentController;
+
+    public ApplicationController(Scanner scanner, ConsoleLogger logger,
+                                 CategoryController categoryController,
+                                 CustomerController customerController,
+                                 DeviceController deviceController,
+                                 RentController rentController) {
+        this.scanner = scanner;
+        this.logger = logger;
+        this.categoryController = categoryController;
+        this.customerController = customerController;
+        this.deviceController = deviceController;
+        this.rentController = rentController;
+    }
+
+    public void mainLoop() {
+        Options option;
+        printOptions();
+        option = readOption();
+        executeOption(option);
+    }
+
+    private void executeOption(Options option) {
+        switch (option) {
+            case ADD_DEVICE -> deviceController.createDevice();
```

Comment:
brakuje break; ten switch bedzie błednie działać.

Reply:
zgodnie ze switch expressions wprowadzonymi od Javy 13, jeśli jest wyrażenie strzałkowe to nie trzeba dawać "break"